shared/provision/code-page/index.tsx

```diff
@@ -20,24 +20,37 @@ type Props = {
   currentDeviceName: string
   otherDeviceName: string
   otherDeviceType: DeviceType
-  tabOverride?: Tab | null
+  tabOverride?: Tab
   textCode: string
   onBack: () => void
   onSubmitTextCode: (code: string) => void
+  setHeaderBackgroundColor: (color: string) => void
 }
 
 type State = {
+  code: string
   tab: Tab
 }
 
 class CodePage2 extends React.Component<Props, State> {
+  static navigationOptions = ({navigation}) => ({
+    header: null,
+    headerBottomStyle: {height: undefined},
+    headerLeft: null,
+    headerStyle: navigation.getParam('headerStyle'),
+  })
   constructor(props: Props) {
     super(props)
     this.state = {
+      code: '',
       tab: (__STORYBOOK__ && this.props.tabOverride) || this._defaultTab(this.props),
     }
   }
 
+  componentDidMount() {
+    this.props.setHeaderBackgroundColor(this._tabBackground())
```

Comment:
Sets `headerStyle` in navigation params so it can be picked up by `navigationOptions` above.